src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -549,90 +627,133 @@ def _get_enable_managed_identity(
-        enable_pod_identity_with_kubenet is not, raise a RequiredArgumentMissingError.
+        specified but enable_pod_identity is, raise a RequiredArgumentMissingError. Will also call function
+        "validate_pod_identity_with_kubenet" for verification. In update mode, if both
+        enable_pod_identity and disable_pod_identity are specified, raise a MutuallyExclusiveArgumentError.
 
         :return: bool
         """
         # read the original value passed by the command
         enable_pod_identity = self.raw_param.get("enable_pod_identity")
-        # try to read the property value corresponding to the parameter from the `mc` object
-        if (
-            self.mc and
-            self.mc.pod_identity_profile and
-            self.mc.pod_identity_profile.enabled is not None
-        ):
-            enable_pod_identity = self.mc.pod_identity_profile.enabled
+        # In create mode, try to read the property value corresponding to the parameter from the `mc` object.
+        if self.decorator_mode == DecoratorMode.CREATE:
+            if (
+                self.mc and
+                self.mc.pod_identity_profile and
+                self.mc.pod_identity_profile.enabled is not None
+            ):
+                enable_pod_identity = self.mc.pod_identity_profile.enabled
 
         # this parameter does not need dynamic completion
         # validation
         if enable_validation:
-            if enable_pod_identity and not self._get_enable_managed_identity(enable_validation=False):
-                raise RequiredArgumentMissingError(
-                    "--enable-pod-identity can only be specified when --enable-managed-identity is specified"
-                )
-            if self.mc and self.mc.network_profile and safe_lower(self.mc.network_profile.network_plugin) == "kubenet":
-                if enable_pod_identity and not self._get_enable_pod_identity_with_kubenet(enable_validation=False):
+            if self.decorator_mode == DecoratorMode.CREATE:
+                if enable_pod_identity and not self._get_enable_managed_identity(enable_validation=False):
                     raise RequiredArgumentMissingError(
-                        "--enable-pod-identity-with-kubenet is required for enabling pod identity addon "
-                        "when using Kubenet network plugin"
+                        "--enable-pod-identity can only be specified when --enable-managed-identity is specified"
+                    )
+                self.validate_pod_identity_with_kubenet(
+                    self.mc,
+                    enable_pod_identity,
+                    self._get_enable_pod_identity_with_kubenet(
+                        enable_validation=False
+                    ),
+                )
+            elif self.decorator_mode == DecoratorMode.UPDATE:
+                if enable_pod_identity and self._get_disable_pod_identity(enable_validation=False):
```

Comment:
nit: we better to extract this check out along with L654

Reply:
also, we are throwing `RequiredArgumentMissingError` in the CRATE code path, while using `MutuallyExclusiveArgumentError ` here

Reply:
Ah, I suppose you might need to take another look. We only have the `disable_pod_identity` available in **update** mode and I suppose it's unnecessary to check the mutual exclusion in **create** mode.

Besides, the two validations are not the same, one for mutual exclusion between `enable_pod_identity` and `disable_pod_identity` in **update** mode, the other for if `enable_pod_identity` is specified, `enable_managed_identity` must be specified too in **create** mode.

Reply:
yes you are right, the diff looks the same before